pkg/cli/admin/mustgather/summary.go

```diff
@@ -0,0 +1,179 @@
+package mustgather
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/go-units"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// PrintBasicClusterState prints a human-readable highlight of some basic information about the openshift cluster that
+// is valuable to every caller of `oc must-gather`.  Even different products.
+// This is NOT a place to add your pet conditions.  We have three places for components to report their errors,
+//  1. clusterversion - shows errors applying payloads to transition versions
+//  2. clusteroperators - shows whether every operand is functioning properly
+//  3. alerts - show whether something might be at risk
+// if you find yourself wanting to add an additional piece of information here, what you really want to do is add it
+// to one of those three spots.  Doing so improves the self-diagnosis capabilities of our platform and lets *every*
+// client benefit.
+func (o *MustGatherOptions) PrintBasicClusterState(ctx context.Context) {
+	fmt.Fprintf(o.RawOut, "When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information.\n")
+
+	clusterVersion, err := o.ConfigClient.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintf(o.RawOut, "error getting cluster version: %v\n", err)
+	}
+	fmt.Fprintf(o.RawOut, "ClusterID: %v\n", clusterVersion.Spec.ClusterID)
+	fmt.Fprintf(o.RawOut, "ClusterVersion: %v\n", humanSummaryForClusterVersion(clusterVersion))
+
+	clusterOperators, err := o.ConfigClient.ConfigV1().ClusterOperators().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(o.RawOut, "error getting cluster operators: %v\n", err)
+	}
+	fmt.Fprintf(o.RawOut, "ClusterOperators:\n")
+	fmt.Fprintf(o.RawOut, humanSummaryForInterestingClusterOperators(clusterOperators)+"\n")
+
+	// TODO gather and display firing alerts
+	fmt.Fprintf(o.RawOut, "\n\n")
+}
+
+// longExistingOperators is a list of operators that should be present on every variant of every platform and have
+// existed for all supported version.  We use this to find when things are missing (like if they fail install).
+var longExistingOperators = []string{
```

Comment:
Should we be able to pull this from the payload? I don't like having this hard coded, and would prefer this to be a fall back (should we not be able to pull a list of what should have been installed with the payload).

Reply:
https://github.com/openshift/must-gather/issues/186#issuecomment-705789781 support this ask.